astropy/units/equivalencies.py

```diff
@@ -505,6 +505,20 @@ def convert_K_to_Jy(x_K):
 
     return [(astrophys.Jy, si.K, convert_Jy_to_K, convert_K_to_Jy)]
 
+def beam_angular_area(beam_area):
+    """
+    Convert between the `beam` unit, which is commonly used to express the area
+    of a radio telescope resolution element, and an area on the sky.
+    """
+    beam = beam_area.to(si.sr).value
```

Comment:
For a direct equivalency, one can just do
```
return [(astrophys.beam, beam_area)]
```

Reply:
My comment about still holds - no need to define conversion functions.

Reply:
I tried this approach, and it didn't work.  This is not a direct equivalency in the sense of 1 beam = 1 sr, it requires functions to be defined.

Reply:
I'm confused: it does need the input area of course, but then it should work fine -- see https://github.com/astropy/astropy/pull/5173#issuecomment-235635143

Also, tests seem to pass with your change!?

Reply:
Yes, the tests pass.  But... hm... the tests did pass, but an interactive example I attempted failed, which is confusing.

More important, though, is that the tests and code were inadequate.  The standard use case is *actually*:
```
x = 5*u.Jy/u.beam
y = x.to(u.MJy/u.sr, u.beam_angular_area(omega_B)
```
which fails locally.